cli/azd/pkg/alphafeatures/alpha_feature_manager.go

```diff
@@ -0,0 +1,91 @@
+package alphafeatures
+
+import (
+	"log"
+	"strings"
+
+	"github.com/azure/azure-dev/cli/azd/pkg/config"
+)
+
+type AlphaFeatureManager struct {
+	configManager   config.UserConfigManager
+	userConfigCache config.Config
+	// used for mocking alpha features on testing
+	alphaFeaturesResolver func() []AlphaFeature
+}
+
+func NewAlphaFeaturesManager(configManager config.UserConfigManager) *AlphaFeatureManager {
+	return &AlphaFeatureManager{
+		configManager: configManager,
+	}
+}
+
+func (m *AlphaFeatureManager) ListFeatures() (map[string]AlphaFeature, error) {
+	result := make(map[string]AlphaFeature)
+
+	var alphaFeatures []AlphaFeature
+	if m.alphaFeaturesResolver != nil {
+		alphaFeatures = m.alphaFeaturesResolver()
+	} else {
+		alphaFeatures = mustUnmarshalAlphaFeatures()
+	}
+
+	for _, aFeature := range alphaFeatures {
+		// cast is safe here from string to AlphaFeatureId
+		status := disabledText
+		if m.IsEnabled(AlphaFeatureId(aFeature.Id)) {
+			status = enabledText
+		}
+
+		result[aFeature.Id] = AlphaFeature{
+			Id:          aFeature.Id,
+			Description: aFeature.Description,
+			Status:      status,
+		}
+	}
+
+	return result, nil
+}
+
+func (m *AlphaFeatureManager) IsEnabled(featureId AlphaFeatureId) bool {
+	if m.userConfigCache == nil {
+		config, err := m.configManager.Load()
+		if err != nil {
+			log.Panic("Can't load user config!! %w", err)
```

Comment:
I think the `configManager` probably has this error context. Maybe attaching the error context for "loading features from config" is more appropriate?

Reply:
Also, I don't disagree with the `panic` pattern here. Over time, I do think we'll start seeing that it's hard to work with a `config` that might or might not be possible to load. And we want to start moving some of these things like "config loading" into an initialization panic when the application starts / command initializes.